### DISCUSSION TITLE
Path to Version 1 - what should the community expect?

### DISCUSSION THREAD

**Discussion by loganvolkers** [chakra-ui/zag#1375] [Q&A] [answered]
Thanks for the great library! I'm the author of some libraries in the state management space, and I think the approach of having a shared core in Zag is far better than copy and paste approaches in Headless UI or Radix. Good work!

The main downside to Zag is that the version number is `0.40.0`. This seems to imply the entire package is beta software. Is that intended to be the signal to the community? What does that mean from a maintenance and breaking changes perspective? Does the lack of a Version 1 mean that the API is expected to be highly unstable?

People on my team have suffered from the Zero branch guarantees in other libraries. For example Slate has made major breaking changes in the Zero branch instead of launching a V2 or V3 when those major breaking changes have come out.

 - Does this package intend to stick to Semver?
 - What maintenance expectations should the community have with the Version Zero branch?

**Comment by segunadebayo** [comment, marked as answer]
Hi @loganvolkers,

Thanks for your kind remark. I truly believe Zag.js' approach is the best for most component libraries/apps.

We're aiming for a v1 in the next month or so following the upcoming Chakra UI v3 release. Afterward, we would commit strictly to semantic versioning.

For the zero branch, it's largely stable. Any breaking changes are clearly documented in the changelog to ensure we're transparent.

**Reply by loganvolkers** [reply]
Thanks @segunadebayo that's great to hear - I added a [recipe for Zag to bunshi](https://www.bunshi.org/recipes/zag/) as my first contribution to the community, and will look at adding some Zag adapters for Jotai and Nanostores as some next contributions.
